### PR TITLE
More pure in chat actions

### PR DESCRIPTION
**Pull request by chrisnojima**
prep for a larger refactor
- [x] use pure more
- [x] use yield select() more

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/actions/chat/attachment.js [hunk 1]
```diff
@@ -418,37 +441,44 @@ function* onRetryAttachment({payload: {message}}: ChatGen.RetryAttachmentPayload
...
   }
+
+  return Saga.sequentially(actions)
 }
 
 function attachmentLoaded(action: ChatGen.AttachmentLoadedPayload) {
   const {payload: {messageKey, path, isPreview}} = action
   if (isPreview) {
-    return Saga.all([
+    return Saga.sequentially([
```
Comment (MarcoPolo) [inline shared/actions/chat/attachment.js]: Did you find that this mattered?

File: shared/actions/chat/attachment.js [hunk 2]
```diff
@@ -539,9 +567,9 @@ function* registerSagas(): SagaGenerator<any, any> {
   )
 
   if (enableActionLogging) {
-    yield Saga.safeTakeEvery(ChatGen.loadAttachmentPreview, _logLoadAttachmentPreview)
-    yield Saga.safeTakeEvery(ChatGen.attachmentLoaded, _logAttachmentLoaded)
-    yield Saga.safeTakeEvery(ChatGen.downloadProgress, _logDownloadProgress)
+    yield Saga.safeTakeEveryPure(ChatGen.loadAttachmentPreview, _logLoadAttachmentPreview)
+    yield Saga.safeTakeEveryPure(ChatGen.attachmentLoaded, _logAttachmentLoaded)
+    yield Saga.safeTakeEveryPure(ChatGen.downloadProgress, _logDownloadProgress)
```
Comment (MarcoPolo) [inline shared/actions/chat/attachment.js]: I'm going to take these out soon

File: shared/actions/chat/attachment.js (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/chat/attachment.js]: I think in practice with just puts it's essentially the same but mike asked about it and I think making it more explicit is likely better.

**Pull request merged**